src/FileFetcher.php

```diff
@@ -218,6 +252,13 @@ private function setState($state)
         $this->getResult()->setData(json_encode($state));
     }
 
+    private function deleteFile($file)
```

Comment:
Really necessary?

Reply:
@dafeder yes. Having that code in the constructor was making our complexity higher than code climate likes.